src/PackageLagMonitor/SearchServiceClient.cs

```
@@ -107,18 +117,26 @@ public Task<SearchResultResponse> GetResultForPackageIdVersion(Instance instance
             RegionInformation regionInformation,
             CancellationToken token)
         {
-            var result = await _azureManagementApiWrapper.GetCloudServicePropertiesAsync(
-                _configuration.Value.Subscription,
-                regionInformation.ResourceGroup,
-                regionInformation.ServiceName,
-                ProductionSlot,
-                token);
+            switch (regionInformation.ServiceType)
+            {
+                case ServiceType.LuceneSearch:
+                    var result = await _azureManagementApiWrapper.GetCloudServicePropertiesAsync(
+                        _configuration.Value.Subscription,
+                        regionInformation.ResourceGroup,
+                        regionInformation.ServiceName,
+                        ProductionSlot,
+                        token);
 
-            var cloudService = AzureHelper.ParseCloudServiceProperties(result);
+                    var cloudService = AzureHelper.ParseCloudServiceProperties(result);
 
-            var instances = GetInstances(cloudService.Uri, cloudService.InstanceCount, regionInformation);
+                    var instances = GetInstances(endpointUri: cloudService.Uri, instanceCount: cloudService.InstanceCount, regionInformation: regionInformation, serviceType: ServiceType.LuceneSearch);
 
-            return instances;
+                    return instances;
+                case ServiceType.AzureSearch:
+                    return GetInstances(endpointUri: new Uri(regionInformation.BaseUrl), instanceCount: 1, regionInformation: regionInformation, serviceType: ServiceType.AzureSearch);
+                default:
+                    throw new UnknownServiceTypeException();
```

You can throw an existing exception like `NotImplementated` with a message containing the service type to avoid the custom exception